benchmark/manyProperties.js

```diff
@@ -0,0 +1,65 @@
+var State = require('../ampersand-state');
+
+// State slows down massively with many properties.
+// We benchmark this here.
+function createModel() {
+  var aModel = State.extend({
+    derived: {
+      foo: {
+        deps: ['a', 'b'],
+        fn: function derived () {
+          return this.a + this.b;
+        }
+      }
+    }
+  });
+  return aModel;
+}
+
+// Go through most of the normal ascii range.
+// Adjust I and J for larger objects.
+var properties = {};
```

Comment:
Thought this was useful to measure, but I haven't been able to move the needle too much on this just yet. The changes above help on the initial set, but by far the most expensive part is enumerating a large object's properties.

Technically, if extraProperties is set to `'ignore'`, we can potentially make a large optimization by defining a `this._defKeys` array, and iterating through that on `set` rather than enumerating the keys of the input object. I see enormous gains on this benchmark when doing so.